src/firefly/java/edu/caltech/ipac/table/ObsCorePackager.java

```diff
@@ -0,0 +1,110 @@
+import edu.caltech.ipac.util.download.URLDownload;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.net.URL;
+
+
+@SearchProcessorImpl(id = "ObsCorePackager")
+public class ObsCorePackager extends FileGroupsProcessor {
+    private static final Logger.LoggerImpl LOGGER = Logger.getLogger();
+
+    public List<FileGroup> loadData(ServerRequest request) throws IOException, DataAccessException {
+        assert (request instanceof DownloadRequest);
+        try {
+            return computeFileGroup((DownloadRequest) request);
+        } catch (Exception e) {
+            LOGGER.error(e);
+            throw new DataAccessException(e.getMessage());
+        }
+    }
+
+
+    private List<FileGroup> computeFileGroup(DownloadRequest request) throws IOException, IpacTableException, DataAccessException {
+        ArrayList<Integer> selectedRows = new ArrayList<>(request.getSelectedRows());
+
+        //todo: keeping these lines in for now, in case we decide to do folders, otherwise these can be removed
+        String zipType = request.getParam("zipType"); //folder or flat
+        boolean doFolders = false;//zipType != null && zipType.equalsIgnoreCase("folder");
+
+        List<FileInfo> fileInfos = new ArrayList<>();
+        MappedData dgDataUrl = EmbeddedDbUtil.getSelectedMappedData(request.getSearchRequest(), selectedRows); //returns all columns
+
+        //todo: using naming scheme from ObsCoreConverter.js getObsCoreRowMetaInfo for now - use productTitleTemplate eventually
+        //todo: for the above, get request.searchRequest.params.template? from the request
+
+        for (int idx : selectedRows) {
+            String access_url = (String) dgDataUrl.get(idx, "access_url");
+            String access_format = (String) dgDataUrl.get(idx, "access_format");
+            String obs_collection = (String) dgDataUrl.get(idx, "obs_collection");
+            String instrument_name = (String) dgDataUrl.get(idx, "instrument_name");
+            String obs_id = (String) dgDataUrl.get(idx, "obs_id");
+            URL url = new URL(access_url);
+            String prepend_file_name = obs_collection + "-" + instrument_name + "-" + obs_id;
+
+            if (access_format.contains("datalink")) {
+                List<FileInfo> tmpFileInfo = parseDatalink(url, prepend_file_name);
+                fileInfos.addAll(tmpFileInfo);
+                continue;
+            }
+
+            String dataPath = url.getFile();
+            String fileName = dataPath;
+            File f = new File(fileName);
+            String extName = f.getName(); //doFolders ? fileName : f.getName();
+            extName = prepend_file_name + "_" + extName;
+
+            FileInfo fileInfo = new FileInfo(access_url, extName, 0);
+            fileInfos.add(fileInfo);
+        }
+
+        HttpServiceInput authCreds = HttpServiceInput.createWithCredential();
+        fileInfos.forEach(fileInfo -> fileInfo.setRequestInfo(authCreds));
```

Comment:
In most cases, `requestUrl` is needed to determine when credential is needed when `createWithCredential(requestUrl)` is called.  Try this...
```
    fileInfos.forEach(fileInfo -> fileInfo.setRequestInfo(HttpServiceInput.createWithCredential(fileInfo.getInternalFilename())));
```